foundationdb/src/directory/node.rs

```diff
@@ -0,0 +1,65 @@
+use crate::tuple::{Subspace};
+use crate::future::FdbSlice;
+use crate::{Transaction, Directory, DirectoryResult};
+use crate::directory::directory_subspace::DirectorySubspaceResult;
+
+
+pub struct Node {
+    subspace: Subspace,
+    path: Vec<String>,
+    target_path: Vec<String>,
+    _layer: Option<Vec<u8>>
```

Comment:
Layer being used as a cache you might want to consider https://docs.rs/futures/0.3.4/futures/future/trait.FutureExt.html#method.shared.
RefCell might also be required in order to not requires `&mut self` on a lot of methods (but be careful about dynamic borrows lifespans in async context).